packages/aws-cdk-lib/core/lib/private/runtime-info.ts

```diff
@@ -106,14 +163,26 @@ export function constructInfoFromStack(stack: Stack): ConstructInfo[] {
 
   addValidationPluginInfo(stack, allConstructInfos);
 
-  // Filter out duplicate values
-  const uniqKeys = new Set();
-  return allConstructInfos.filter(construct => {
-    const constructKey = `${construct.fqn}@${construct.version}`;
-    const isDuplicate = uniqKeys.has(constructKey);
-    uniqKeys.add(constructKey);
-    return !isDuplicate;
+  // Filter out duplicate values and append the metadata information to the array
+  const uniqueMap = new Map<string, ConstructInfo>();
+  allConstructInfos.forEach(info => {
+    const key = `${info.fqn}@${info.version}`;
+    if (uniqueMap.has(key)) {
+      const existingInfo = uniqueMap.get(key);
+      if (existingInfo && existingInfo.metadata && info.metadata) {
+        existingInfo.metadata.push(...info.metadata);
+      }
+    } else {
+      uniqueMap.set(key, info);
+    }
   });
+
+  return Array.from(uniqueMap.values());
+}
+
+function isResource(construct: IConstruct): construct is Resource {
```

Comment:
please add some docs to explain this function

Reply:
also what is the difference between this function, and this one https://github.com/aws/aws-cdk/blob/a2c633f1e698249496f11338312ab42bd7b1e4f0/packages/aws-cdk-lib/core/lib/resource.ts#L126

Reply:
It's the same and duplicated function, but we can't directly use this function as it cause circular dependencies.

Reply:
Will add docstring